mafiasi/gprot/templates/gprot/index.html

```diff
@@ -66,4 +110,31 @@
     });
     $(".help-block").remove();
 </script>
+<script type="text/javascript">
+    function favorite(action){
+        let url = window.location.href
+        $.ajax({
+            type: 'POST',
+            url: '/gprot/favorite/',
+            headers: {'X-CSRFTOKEN': readCookie('csrftoken')},
+            dataType: 'text',
+            data: {'action': action, 'url': url},
+            success: function() {
+                location.reload();
+            }
+        });
+    }
+
+    /** from https://stackoverflow.com/questions/1599287/create-read-and-erase-cookies-with-jquery#1599291 **/
+    function readCookie(name) {
+        let nameEQ = name + "=";
+        let ca = document.cookie.split(';');
+        for (let i = 0; i < ca.length; i++) {
+            let c = ca[i];
+            while (c.charAt(0) === ' ') c = c.substring(1, c.length);
+            if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length);
+        }
+        return null;
+    }
+</script>
```

Comment:
This should also be moved to `static/js/gprot.js`. By the way, you can generate short urls to StackOverflow answers by clicking on Share